### PR TITLE
Bump Roslyn to 3.4.0-beta2-19476-01

### PR DESCRIPTION
**Pull request by sandyarmstrong**
Changes since [2f21c63](https://www.github.com/dotnet/roslyn/commit/2f21c63):
- [Merge pull request #38839 from dotnet/dev/jorobich/update-vs-version-docs](https://www.github.com/dotnet/roslyn/pull/38839)
- [Merge pull request #38831 from genlu/TargetTypeIcon](https://www.github.com/dotnet/roslyn/pull/38831)
- [Merge pull request #38835 from mavasani/Fix_RPS_Regression](https://www.github.com/dotnet/roslyn/pull/38835)
- [Fix TakeIncrementalSnapshot crash (#38752)](https://www.github.com/dotnet/roslyn/pull/38752)
- [Merge pull request #38624 from CyrusNajmabadi/callerSymbol](https://www.github.com/dotnet/roslyn/pull/38624)
- [Merge pull request #38669 from dustincoleman/objectFavorites](https://www.github.com/dotnet/roslyn/pull/38669)
- [Copy value from field on MarshalByRef object to call ToString() (#38772)](https://www.github.com/dotnet/roslyn/pull/38772)
- [Spelling fixes (#38721)](https://www.github.com/dotnet/roslyn/pull/38721)
- [Merge pull request #38789 from jaredpar/fix-xunit](https://www.github.com/dotnet/roslyn/pull/38789)
- [Add tests for method type inference (#38710)](https://www.github.com/dotnet/roslyn/pull/38710)
- [Formatter throws a exception where HostObjectValue is null (#38430)](https://www.github.com/dotnet/roslyn/pull/38430)
- [Merge pull request #38324 from dibarbet/remove_liveshare_ivt](https://www.github.com/dotnet/roslyn/pull/38324)
- [Merge pull request #38799 from genlu/AddTelemetry](https://www.github.com/dotnet/roslyn/pull/38799)
- [Merge pull request #38787 from dibarbet/move_type_to_file_selection](https://www.github.com/dotnet/roslyn/pull/38787)
- [Allow conversion of collection initializer Add extension method this arg (#38732)](https://www.github.com/dotnet/roslyn/pull/38732)
- [Merge pull request #38754 from genlu/FixMoveNamespace](https://www.github.com/dotnet/roslyn/pull/38754)
- [Add a comment explaining the recently added cycle detection code and its relation to the language specification. (#38784)](https://www.github.com/dotnet/roslyn/pull/38784)
- [Verify semantic model on suppression operator (#38756)](https://www.github.com/dotnet/roslyn/pull/38756)
- [Merge pull request #38364 from heejaechang/typscriptSyntacticColorizer](https://www.github.com/dotnet/roslyn/pull/38364)
- [Refine diagnostic for switch expression that is not exhaustive (#38714)](https://www.github.com/dotnet/roslyn/pull/38714)
- [Test constant folding of shift operators. (#38729)](https://www.github.com/dotnet/roslyn/pull/38729)
- [Adjust syntax disambiguation for ref ternary. (#38713)](https://www.github.com/dotnet/roslyn/pull/38713)
- [Add Containing Type and Containing Member columns to FAR (#37802)](https://www.github.com/dotnet/roslyn/pull/37802)
- [Merge pull request #38768 from kendrahavens/patch-1](https://www.github.com/dotnet/roslyn/pull/38768)
- [Merge pull request #38593 from JoeRobich/make-organize-imports-public](https://www.github.com/dotnet/roslyn/pull/38593)
- [Merge pull request #38194 from sharwell/roslyn-etw](https://www.github.com/dotnet/roslyn/pull/38194)
- [Merge pull request #38751 from jnm2/better_rename_test_message](https://www.github.com/dotnet/roslyn/pull/38751)
- [Update immutable arrays of symbols in the NullabilityRewriter (#38736)](https://www.github.com/dotnet/roslyn/pull/38736)
- [Remove legacy completion (#38569)](https://www.github.com/dotnet/roslyn/pull/38569)
- [Add cycle detection for more scenarios involving looking in base interfaces. (#38738)](https://www.github.com/dotnet/roslyn/pull/38738)
- [Merge pull request #38753 from JoeRobich/use-contenttype-isoftype](https://www.github.com/dotnet/roslyn/pull/38753)
- [Merge pull request #38749 from dotnet/dev/jorobich/fix-code-cleanup](https://www.github.com/dotnet/roslyn/pull/38749)
- [Detect when type/field declared in an interface shadows a member from base interface. (#38731)](https://www.github.com/dotnet/roslyn/pull/38731)
- [Add handling for unconventional implementation of events and properties when Default Interface Implementations are involved. (#38683)](https://www.github.com/dotnet/roslyn/pull/38683)
- [Merge pull request #38739 from genlu/RemoveSuppression](https://www.github.com/dotnet/roslyn/pull/38739)
- [Merge pull request #38695 from dibarbet/brace_completion_null](https://www.github.com/dotnet/roslyn/pull/38695)
- [Merge pull request #38692 from jcagme/azdoPublishing](https://www.github.com/dotnet/roslyn/pull/38692)
- [Merge pull request #38688 from jaredpar/fix-sync](https://www.github.com/dotnet/roslyn/pull/38688)
- [Adjust rules for nullable attributes (#38610)](https://www.github.com/dotnet/roslyn/pull/38610)
- [Merge pull request #37629 from lavanyaguru/MakeFuntionStatic](https://www.github.com/dotnet/roslyn/pull/37629)
- [Perform lookup in base interfaces even in a type-only context. (#38564)](https://www.github.com/dotnet/roslyn/pull/38564)
- [Don't warn about empty initializers on nullable properties in object initializers (#38642)](https://www.github.com/dotnet/roslyn/pull/38642)
- [Reuse previously resolved missing assembly metadata objects when identity doesn't match (#38550)](https://www.github.com/dotnet/roslyn/pull/38550)
- [Merge pull request #38703 from mavasani/BumpUpFlowAnalysisUtilitiesVersion](https://www.github.com/dotnet/roslyn/pull/38703)
- [Update dependencies from https://github.com/dotnet/arcade build 20190913.3 (#38696)](https://www.github.com/dotnet/roslyn/pull/38696)
- [Merge pull request #37148 from lgolding/users/lgolding/sarif-v2.1.0](https://www.github.com/dotnet/roslyn/pull/37148)
- [Merge pull request #34617 from sharwell/unused-field](https://www.github.com/dotnet/roslyn/pull/34617)
- [Merge pull request #38551 from sharwell/reference-assemblies](https://www.github.com/dotnet/roslyn/pull/38551)
- [Fix crash when Nullable<T> creation takes dynamic argument (#38647)](https://www.github.com/dotnet/roslyn/pull/38647)
- [Merge pull request #38682 from sharwell/format-solution](https://www.github.com/dotnet/roslyn/pull/38682)
- [Re-infer unary and binary operators in NullableWalker (#38320)](https://www.github.com/dotnet/roslyn/pull/38320)
- [Update dependencies from https://github.com/dotnet/arcade build 20190912.5 (#38681)](https://www.github.com/dotnet/roslyn/pull/38681)
- [Merge pull request #38666 from mavasani/PerfFixes_SymbolUsageAnalysis](https://www.github.com/dotnet/roslyn/pull/38666)
- [Merge pull request #38656 from ryzngard/issues/jaws_inline_rename](https://www.github.com/dotnet/roslyn/pull/38656)
- [Spelling fixes (#38673)](https://www.github.com/dotnet/roslyn/pull/38673)
- [A switch expression only has a common type if all arms can be converted to that type. (#38232)](https://www.github.com/dotnet/roslyn/pull/38232)
- [Merge pull request #38648 from JoeRobich/access-fixes](https://www.github.com/dotnet/roslyn/pull/38648)
- [Merge pull request #38655 from genlu/FixLeak](https://www.github.com/dotnet/roslyn/pull/38655)
- [Update VS SDK to 16.3.2099 (#38472)](https://www.github.com/dotnet/roslyn/pull/38472)
- [Update dependencies from https://github.com/dotnet/arcade build 20190911.7 (#38667)](https://www.github.com/dotnet/roslyn/pull/38667)
- [Merge pull request #38654 from JoeRobich/update-for-snap](https://www.github.com/dotnet/roslyn/pull/38654)
- [Merge pull request #4 from lavanyaguru/codefix](https://www.github.com/dotnet/roslyn/pull/4)
- [Merge pull request #3 from lavanyaguru/codefix](https://www.github.com/dotnet/roslyn/pull/3)

(https://github.com/dotnet/roslyn/compare/2f21c63...5ca9538)

### CODE REVIEW

File: main/src/addins/CSharpBinding/CSharpBinding.csproj [hunk 1]
```diff
@@ -252,20 +252,6 @@
     <Compile Include="MonoDevelop.CSharp.Navigation\Counters.cs" />
     <Compile Include="MonoDevelop.CSharp.Project\CSharpLanguageVersionHelper.cs" />
     <Compile Include="MonoDevelop.CSharp\DocumentContextExtension.cs" />
-    <Compile Include="MonoDevelop.Ide.Completion.Presentation\RoslynCompletionPresenter.cs" />
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/CSharpBinding.csproj]: Old completion is gone, right?

Comment (sandyarmstrong) [reply main/src/addins/CSharpBinding/CSharpBinding.csproj]: Roslyn's old completion provider API, yes. Legacy editor's regular C# support wasn't using it, just these bits for old web editors.

**Comment by sandyarmstrong** [comment]
@Therzok jfyi, I'm holding back on `System.Collections.Immutable` bump until _next_ Roslyn insertion, to align better with what's going on in VSwin.

File: main/src/addins/CSharpBinding/CSharpBinding.csproj (hunk 1 above)
Comment (sandyarmstrong) [reply main/src/addins/CSharpBinding/CSharpBinding.csproj]: Except according to compat check, looks like some of this stuff (contained document) is still in use, so I need to bring that back.

Comment (KirillOsenkov) [reply main/src/addins/CSharpBinding/CSharpBinding.csproj]: Yes we still use it. Prob. need to sync with the latest Roslyn source at some point.

**Pull request merged**